sdk/client/src/broadcast_client.rs

```diff
@@ -0,0 +1,216 @@
+        address: AccountAddress,
+    ) -> Result<Vec<Response<Option<AccountView>>>> {
+        let futures = self
+            .random_clients()
+            .map(|client| client.get_account(address));
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    pub async fn get_transactions(
+        &self,
+        start_seq: u64,
+        limit: u64,
+        include_events: bool,
+    ) -> Result<Vec<Response<Vec<TransactionView>>>> {
+        let futures = self
+            .random_clients()
+            .map(|client| client.get_transactions(start_seq, limit, include_events));
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    pub async fn get_account_transaction(
+        &self,
+        address: AccountAddress,
+        seq: u64,
+        include_events: bool,
+    ) -> Result<Vec<Response<Option<TransactionView>>>> {
+        let futures = self
+            .random_clients()
+            .map(|client| client.get_account_transaction(address, seq, include_events));
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    pub async fn get_account_transactions(
+        &self,
+        address: AccountAddress,
+        start_seq: u64,
+        limit: u64,
+        include_events: bool,
+    ) -> Result<Vec<Response<Vec<TransactionView>>>> {
+        let futures = self.random_clients().map(|client| {
+            client.get_account_transactions(address, start_seq, limit, include_events)
+        });
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    pub async fn get_events(
+        &self,
+        key: &str,
+        start_seq: u64,
+        limit: u64,
+    ) -> Result<Vec<Response<Vec<EventView>>>> {
+        let futures = self
+            .random_clients()
+            .map(|client| client.get_events(key, start_seq, limit));
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    pub async fn get_currencies(&self) -> Result<Vec<Response<Vec<CurrencyInfoView>>>> {
+        let futures = self.random_clients().map(|client| client.get_currencies());
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    pub async fn get_network_status(&self) -> Result<Vec<Response<u64>>> {
+        let futures = self
+            .random_clients()
+            .map(|client| client.get_network_status());
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    //
+    // Experimental APIs
+    //
+
+    pub async fn get_state_proof(&self, from_version: u64) -> Result<Vec<Response<StateProofView>>> {
+        let futures = self
+            .random_clients()
+            .map(|client| client.get_state_proof(from_version));
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    pub async fn get_account_state_with_proof(&self,
```

Comment:
The formatting on these looks really weird, maybe you need to run rustfmt on it?